C/CUDA/CUDA_Runtime/platform_augmentation.jl

```diff
@@ -0,0 +1,78 @@
+using Libdl
+
+using CUDA_Driver
```

Comment:
This makes me nervous... The augmentation hook will execute during Pkg and the environment at that time is unbaked.
So using a non-stdlib dependency can easily break. (Try in an empty depot)

Reply:
```
ERROR: LoadError: ArgumentError: Package CUDA_Driver [a6be3551-5000-4724-bfab-551d491a12f7] is required but does not seem to be installed:
 - Run `Pkg.instantiate()` to install all recorded dependencies.

Stacktrace:
 [1] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1306
 [2] _require_prelocked(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1200
 [3] macro expansion
   @ ./loading.jl:1180 [inlined]
 [4] macro expansion
   @ ./lock.jl:223 [inlined]
 [5] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1144
 [6] include(fname::String)
   @ Base.MainInclude ./client.jl:476
 [7] top-level scope
   @ /tmp/jl_OiUFql/packages/CUDA_Runtime_jll/TB1Nf/.pkg/select_artifacts.jl:4
 [8] include(fname::String)
   @ Base.MainInclude ./client.jl:476
 [9] top-level scope
   @ none:5
in expression starting at /tmp/jl_OiUFql/packages/CUDA_Runtime_jll/TB1Nf/.pkg/platform_augmentation.jl:3
in expression starting at /tmp/jl_OiUFql/packages/CUDA_Runtime_jll/TB1Nf/.pkg/select_artifacts.jl:4
```